src/pushsource/_impl/backend/registry_source.py

```diff
@@ -0,0 +1,171 @@
+import logging
+
+
+try:
+    from urllib.parse import urlparse
+except ImportError: # pragma: no cover
+    from urlparse import urlparse
+
+from ..source import Source
+from ..model import (
+    ContainerImagePushItem,
+    SourceContainerImagePushItem,
+    ContainerImagePullInfo,
+    ContainerImageTagPullSpec,
+    ContainerImageDigestPullSpec,
+)
+from ..utils.containers import (
+    get_manifest,
+    inspect,
+    MT_S2_V2,
+    MT_S2_V1,
+    MT_S2_V1_SIGNED,
+    MT_S2_LIST,
+)
+
+from ..helpers import list_argument
+
+LOG = logging.getLogger("pushsource")
+
+
+class RegistrySource(Source):
+    """Uses URIs of container images as source for push items."""
+
+    def __init__(
+        self,
+        dest=None,
+        registry_uris=None,
+        signing_key=None,
+        docker_url="unix://var/run/docker.sock",
+        executor_container_image=None,
+        docker_timeout=None,
+        docker_verify_tls=None,
+        docker_cert_path=None,
+    ):
+        """Create a new source.
+
+        Parameters:
+            dest (str, list[str])
+                The destination(s) to fill in for push items created by this
+                source. If omitted, all push items have empty destinations.
+
+            registry_uris (list[str])
+                List of references to container images with tags+dest tags
+                Format <host>/<namespace>/<repo>:<tag>:<destination_tag>:<destination_tag>
+                Example: registry.redhat.io/ubi:8:latest:8:8.1
+
+            signing_key (list[str])
+                GPG signing key ID(s). If provided, content must be signed
+                using one of the provided keys. Include ``None`` if unsigned
+                should also be permitted.
+                Keys should be listed in the order of preference.
+
+            executor_container_image (str):
+                Path to the container image in which to execute the commands. Must be downloadable
+                without extra permissions.
+
+            docker_url (str):
+                URL of the docker client that should run the container. Local socket by default.
+
+            docker_timeout (int):
+                Timeout for executing Docker commands. Disabled by default.
+
+            docker_verify_tls (bool):
+                Whether to perform TLS verification with the Docker client. Disabled by default.
+
+            docker_cert_path (str):
+                Path to Docker certificates for TLS authentication. '~/.docker' by default.
+        """
+        self._registry_uris = registry_uris
+        self._dest = list_argument(dest)
+        self.signing_keys = list_argument(signing_key)
```

Comment:
Could you please name all these attributes with a leading `_` to signal that they're private to this class?